frontend/src/component/commandBar/RecentlyVisited/RecentlyVisited.tsx

```diff
@@ -37,55 +39,127 @@ const StyledListItemText = styled(ListItemText)(({ theme }) => ({
-            };
-        }
-        if (item.projectId) {
-            return {
-                name: item.projectId,
-                path: `/projects/${item.projectId}`,
-                icon: <StyledProjectIcon />,
-            };
-        }
-        return {
-            name: item.featureId ?? item.projectId ?? item.pathName,
-            path: item.pathName || '/',
-            icon: <IconRenderer path={item.pathName ?? '/unknown'} />,
-        };
-    });
+const toListItemButton = (
+    item: LastViewedPage,
+    routes: Record<string, { path: string; route: string; title: string }>,
+    index: number,
+) => {
+    const key = `recently-visited-${index}`;
+    if (item.featureId && item.projectId) {
+        return (
+            <RecentlyVisitedFeatureButton
+                key={key}
+                featureId={item.featureId}
+                projectId={item.projectId}
+            />
+        );
+    }
+    if (item.projectId) {
+        return (
+            <RecentlyVisitedProjectButton
+                key={key}
+                projectId={item.projectId}
+            />
+        );
+    }
+    if (!item.pathName) return null;
+    const name = routes[item.pathName]?.title ?? item.pathName;
+    return (
+        <RecentlyVisitedPathButton key={key} path={item.pathName} name={name} />
+    );
+};
+
+const RecentlyVisitedFeatureButton = ({
+    key,
+    projectId,
+    featureId,
+}: { key: string; projectId: string; featureId: string }) => {
+    return (
+        <ListItemButton
+            key={key}
+            dense={true}
+            component={Link}
+            to={`/projects/${projectId}/features/${featureId}`}
+            sx={listItemButtonStyle}
+        >
+            <StyledListItemIcon>
+                <Icon>{'flag'}</Icon>
+            </StyledListItemIcon>
+            <StyledListItemText>
+                <Typography color='textPrimary'>{featureId}</Typography>
+            </StyledListItemText>
+        </ListItemButton>
+    );
+};
+
+const RecentlyVisitedPathButton = ({
+    path,
+    key,
+    name,
+}: { path: string; key: string; name: string }) => {
+    return (
+        <ListItemButton
+            key={key}
+            dense={true}
+            component={Link}
+            to={path}
+            sx={listItemButtonStyle}
+        >
+            <StyledListItemIcon>
+                <ConditionallyRender
+                    condition={path === '/projects'}
+                    show={<StyledProjectIcon />}
+                    elseShow={<IconRenderer path={path} />}
+                />
+            </StyledListItemIcon>
+            <StyledListItemText>
+                <Typography color='textPrimary'>{name}</Typography>
+            </StyledListItemText>
+        </ListItemButton>
+    );
+};
+
+const RecentlyVisitedProjectButton = ({
+    projectId,
+    key,
+}: { projectId: string; key: string }) => {
+    const { project, loading } = useProjectOverview(projectId);
```

Comment:
Hmm downside of this is that every time they open the menu, we will make 1-5 concurrent calls to backend. But for now it is fine.

Reply:
Yup, unfortunately